### DISCUSSION TITLE
Deployed succesfully on zkSync but not seen in broadcast folder

### DISCUSSION THREAD

**Discussion by AlimGlck** [Cyfrin/foundry-full-course-cu#2587] [Q&A]
As you see it doesn't give any error but doesn't show also in the broadcast folder.

![image](https://github.com/user-attachments/assets/f8d7530f-f8ae-4ba3-9094-38feed9937fb)

Is it a bug or something, Do you have any ideas?

**Comment by Niferu** [comment]
Hi @AlimGlck 

Try adding `--broadcast` flag and if that did not help just remove whole broadcast folder as it is currently bugging out very often if you are changing names of your contracts

**Reply by AlimGlck** [reply]
It doesn't accept the ``` --broadcast ``` variable on command and when I delete the broadcast folder and deploy it again it doesn't even create a broadcast folder again. So I deleted all folders and did it from the start but didn't work again. I guess it's best to leave it like that.

**Reply by EngrPips** [reply]
I did `forge script` recently, and the `broadcast` folder was created. Maybe try with `forge script` and let's see if that generates the `broadcast` folder.

**Reply by AlimGlck** [reply]
You mean instead of ```forge create``` right?

**Reply by EngrPips** [reply]
That won't work on `zksync` actually, I remember, arrgh.

**Reply by EngrPips** [reply]
You can just leave it as I tried, too, and I was unable to generate the `broadcast` folder by deploying to `zksync` chain. If you deploy to sepolia, it will be a different case.

**Reply by AlimGlck** [reply]
Okay, I will try the last time if doesn't work, I'll ignore it.

**Reply by AlimGlck** [reply]
It doesn't work and the docs in the video and the current docs are not the same either. So maybe we need to do it differently right now or maybe the problem is with the docker. I checked the foundry-zksync-book but nothing helpful there as well.

**Reply by EngrPips** [reply]
arrgh, Maybe something changed. Maybe I can check the docs too and see if I find anything Interesting.

**Reply by Niferu** [reply]
Hmm thats very strange as I was working with AlephZero blockchain recently (deploy is available by `create` only) and broadcast folder works perfectly fine, so maybe foundry team needs to map zkSync somewhere to get it working as I know new blockchains reach to foundry team to include them in some mappings etc. Thats why we can use `script` instead of `create` for example

**Comment by EngrPips** [comment]
Hello @AlimGlck, Please do as @Nlferu, and let's see how it will play out.

**Reply by EngrPips** [reply]
I am looking forward to hearing how it played out.

**Comment by tcrin** [comment]
Hi Mr. @AlimGlck ,

I’ve faced a similar issue before! 💡

The **broadcast** folder is only created when you use the `--broadcast` option in the `forge script` command. Even though Mr. @PatrickAlphaC mentioned that the `forge script` command is not well supported on zkSync, that doesn't mean we can't use it 😄.

🛠 Try running:
```bash
forge script script/DeploySimpleStorage.s.sol --rpc-url=$ZKSYNC_RPC_URL --private-key=$ZKSYNC_PRIVATE_KEY --legacy --zksync --broadcast
```

🔍 Note: Make sure to replace **script/DeploySimpleStorage.s.sol** with the correct path to your script!

After the script run successfully, **broadcast** folder will be created.

**Comment by EngrPips** [comment]
Interesting
